utils_test/__init__.py: rewrite get_free_space()

PandaWei opened the pull request:
The codes of get_free_space call self.runner.run twice,
and No catch IndexError/ValueError for the following,
    int(line.split()[3]) / 1048576
Besides above, It's better to output the return of
self.runner.run for failure case, not for successful case.
and so on.

So need to rewirte and make it more better.

yangdongsheng commented on virttest/utils_test/__init__.py:
```diff
@@ -1802,26 +1802,35 @@ def get_free_space(self, disk_type, path='/', vgname=None):
+            else:
+                raise error.TestError("Get %s space failed: %s." %
+                                      (vgname, output))
+        return avail_size
```
What's the unit of the ret-value.

PandaWei replied:
It's 'G'

yangdongsheng replied:
Okey, maybe a comment for this function would help a lot. And then you can use "df -BG" to get the result in GiB unit. In addition, is there any bug in the original implementaion? Or that's just a cleanup again?

PandaWei replied:
yes, it's just clean up.
I will update it as your comments.

PandaWei commented:
1) fix bug(s) for 'lvm' type when match free space
   - call .stdout twice that leads to no attribute 'stdout'
     output = self.runner.run("vgs --units=g | grep %s" % vgname).stdout
     if re.search(vgname, output.stdout):
     (Pdb) output.stdout
     *** AttributeError: 'str' object has no attribute 'stdout'

   - Old expression doesn't get correct space number, as following,
     [root@localhost tp-libvirt]# vgs --units=g
      VG     #PV #LV #SN Attr   VSize  VFree
     fedora   1   2   0 wz--n- 49.15g    0g
     [root@localhost tp-libvirt]# python
     >>> output="  fedora   1   2   0 wz--n- 49.15g    0g"
     >>> output.split('g')[0]
     '  fedora   1   2   0 wz--n- 49.15'
2) Using option '-BG' to get space directly by G.
   Old code doesn't catch IndexError for 'int(line.split()[3])/1048576'.
3) add explanation for the unit of return value.
4) cleanup codes and avoid call twice for self.runner.run().

yangdongsheng commented on virttest/utils_test/__init__.py:
```diff
@@ -1800,28 +1800,39 @@ def __init__(self, params):
+                                      (os.path.dirname(path), str(detail)))
         elif disk_type == "lvm":
-            output = self.runner.run("vgs --units=g | grep %s" % vgname).stdout
-            if re.search(vgname, output.stdout):
```
That's really a bug, because the output is a str rather than a Result object, then there is no .stdout in output. Good catch!!.

yangdongsheng commented:
Thanx for your summary of your patch, that would help a lot to reviewer. And your patch looks much better in this version. Good job.

yangdongsheng commented:
If you past the result of the cases which are affected by this patch, I think we can let it go.  I believe there is a case which failed would pass after this commit.

PandaWei commented:
@yangdongsheng
Thanks for your comments，
In fact, I haven't no case failed by that problem.
Initially, I just want to cleanup code and pay close attention to file type, not lvm type.

You asked me to do more tests about my patch.
so i wrote some small scripts to call this function and verify it.
in this process, i found the problem of lvm.

so thanks to your reminder， i found this bug。

will-Do commented:
Okay, good fix.
Now I think we can merge this patch.
